packages/eslint-plugin/src/rules/no-unused-private-class-members.ts

```diff
@@ -0,0 +1,261 @@
+        return;
+      }
+
+      // Any assignments to this member, except for assignments that also read
+      if (isWriteOnlyAssignment(node)) {
+        return;
+      }
+
+      const wrappingExpressionType = node.parent.parent?.type;
+      const parentOfWrappingExpressionType = node.parent.parent?.parent?.type;
+
+      // A statement which only increments (`this.#x++;`)
+      if (
+        wrappingExpressionType === AST_NODE_TYPES.UpdateExpression &&
+        parentOfWrappingExpressionType === AST_NODE_TYPES.ExpressionStatement
+      ) {
+        return;
+      }
+
+      /*
+       * ({ x: this.#usedInDestructuring } = bar);
+       *
+       * But should treat the following as a read:
+       * ({ [this.#x]: a } = foo);
+       */
+      if (
+        wrappingExpressionType === AST_NODE_TYPES.Property &&
+        parentOfWrappingExpressionType === AST_NODE_TYPES.ObjectPattern &&
+        node.parent.parent?.value === node.parent
+      ) {
+        return;
+      }
+
+      // [...this.#unusedInRestPattern] = bar;
+      if (wrappingExpressionType === AST_NODE_TYPES.RestElement) {
+        return;
+      }
+
+      // [this.#unusedInAssignmentPattern] = bar;
+      if (wrappingExpressionType === AST_NODE_TYPES.ArrayPattern) {
+        return;
+      }
+
+      // We can't delete the memberDefinition, as we need to keep track of which
+      // member we are marking as used. In the case of nested classes, we only
+      // mark the first member we encounter as used. If you were to delete the
+      // member, then any subsequent usage could incorrectly mark the member of
+      // an encapsulating parent class as used, which is incorrect.
+      trackedClassMembersUsed.add(memberDefinition.declaredNode);
+    }
+
+    return {
+      // Collect all declared members/methods up front and assume they are all
+      // unused.
+      ClassBody(classBodyNode): void {
+        const privateMembers = new Map<string, PrivateMember>();
+
+        trackedClassMembers.unshift(privateMembers);
+        for (const bodyMember of classBodyNode.body) {
+          if (
+            (bodyMember.type === AST_NODE_TYPES.PropertyDefinition ||
+              bodyMember.type === AST_NODE_TYPES.MethodDefinition) &&
+            (bodyMember.key.type === AST_NODE_TYPES.PrivateIdentifier ||
+              (bodyMember.key.type === AST_NODE_TYPES.Identifier &&
+                bodyMember.accessibility === 'private'))
+          ) {
+            privateMembers.set(bodyMember.key.name, {
+              declaredNode: bodyMember,
+              isAccessor:
+                bodyMember.type === AST_NODE_TYPES.MethodDefinition &&
+                (bodyMember.kind === 'set' || bodyMember.kind === 'get'),
+            });
+          }
+        }
+      },
+
+      // Process nodes like:
+      // ```ts
+      // class A {
+      //   #myPrivateMember = 123;
+      // }
+      // ```
+      PrivateIdentifier(node): void {
+        processPrivateIdentifier(node);
```

Comment:
Threading https://github.com/typescript-eslint/typescript-eslint/pull/10265#issuecomment-2452975351: I don't get that error locally. But I do get what Nx is getting (https://cloud.nx.app/runs/lsSb3ZQa0g?utm_source=pull-request&utm_medium=comment):

```plaintext
src/rules/no-unused-private-class-members.ts(215,34): error TS2345: Argument of type 'PrivateIdentifier' is not assignable to parameter of type 'Identifier'.
  Type 'PrivateIdentifier' is missing the following properties from type 'Identifier': decorators, optional, typeAnnotation
```

Maybe a caching issue? I'd suggest merging from `main` and re-building. 🤷

Reply:
I merged with main and now I am getting this Netlify error: https://app.netlify.com/sites/typescript-eslint/deploys/67295a26e190f200080d6053

Reply:
Furthermore, it looks like "docs.test.ts" has a lint error, but that is unrelated to my PR.

Reply:
#10288 ✨

Reply:
I merged with main, but CI is now getting this error:
```
- 429: Unable to connect to Nx Cloud.
```

Reply:
[429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429)

![Screenshot of nine commits fro Zamiell, either named 'update' or a merge from main](https://github.com/user-attachments/assets/5b93d4a1-7c1b-4b1f-b545-9334a1dd6847)

It'll probably ease up on you if you push less.

Reply:
If you are implying that I am committing too much, I thought it was best-practice to put each PR fix that was brought up in a PR review thread by other people into a separate commit, which I did.

Anyways, I just remerged from main again and am still getting these two CI errors:
- https://cloud.nx.app/runs/WZCMfyPxRY/task/typescript-eslint%3Atest
- https://cloud.nx.app/runs/KCqe66ahXv/task/eslint-plugin%3Atest

Not sure what they mean though.

Reply:
> best-practice to put each PR fix that was brought up in a PR review thread by other people into a separate commit

I don't subscribe to that best practice 😄. It can be nice to have granular commits when there are tricky things being discussed. But there's no real utility most of the time.

The downside is that a lot of CI churn gets added to the repo. Which is never good: almost all repos are either on a rate-limited free plan (like we are) or a pay-for-bandwidth plan ($$$).

> Now, I get these two CI errors:

You've got some test failures, have you tried reading through the test files mentioned in them? I haven't looked deeply but if you've got specific questions I can.

Reply:
> You've got some test failures

Right, but the part I am confused about is that the test failures are in files that I haven't modified. Do I have to add the new rule to one of the existing configs or something

Reply:
That's what I'm trying to hint at with my questions: did you read the tests that are failing? The test names, the test failures, the code they touch?

I'm not intentionally trying to be obtuse or unhelpful. But these are things that I think as someone who's got a bunch of PRs merged into this repo (❤️) you're equipped to debug.

Reply:
That's what I'm trying to hint at with my questions: did you read the tests that are failing? The test names, the test failures, the code they touch? 

Ideally, the repo's docs and tests should be set up so that you can debug through these and figure this out on your own. If it's not clear from them then that would be something we'd want to work on. But we can't tell whether the repo's setup is at fault without an honest try.

I'm not intentionally trying to be obtuse or unhelpful. But these are things that I think as someone who's got a bunch of PRs merged into this repo (❤️) you're equipped to debug.

Reply:
That's what I'm trying to hint at with my questions: did you read the tests that are failing? The test names, the test failures, the code they touch? 

Ideally, the repo's docs and tests should be set up so that you can debug through these and figure this out on your own. If it's not clear from them then that would be something we'd want to work on.

I'm not intentionally trying to be obtuse or unhelpful. But these are things that I think as someone who's got a bunch of PRs merged into this repo (❤️) you're equipped to debug.